src/adapter.js

```diff
@@ -157,15 +165,22 @@ function makeSourceMap(SourceMap, ast, element, path) {
 }
 
 // Make a new annotation for the given path and message
-function makeAnnotation(Annotation, SourceMap, ast, result, info, path, message) {
+function makeAnnotation(Annotation, Link, SourceMap, ast, result, info, path, message) {
   const annotation = new Annotation(message);
   annotation.classes.push(info.type);
   annotation.code = info.code;
   result.content.push(annotation);
 
+  if (info.relation) {
+    const link = new Link();
+    link.relation = 'origin';
+    link.href = `http://docs.apiary.io/validations/swagger#${info.relation}`;
+    annotation.links.push(link);
+  }
+
   if (ast && path) {
     const position = getPosition(ast, path);
-    if (position) {
+    if (position && position.start !== undefined && !isNaN(position.start) && position.end !== undefined && !isNaN(position.end)) {
```

Comment:
Do you need the `!== undefined` checks? Doesn't `!isNaN(undefined)` return false?